Spigot-API-Patches/0276-Extend-hide-showPlayer-API.patch

```diff
@@ -0,0 +1,86 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Tom <cryptite@gmail.com>
+Date: Fri, 26 Feb 2021 16:36:59 -0600
+Subject: [PATCH] Extend hide/showPlayer API
+
+
+diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
+index 5fe48dcc5a3ea492915350cff44d5f112ce11dbf..8f2869e3dc695fb73bb7edb2390ae6b8e788e406 100644
+--- a/src/main/java/org/bukkit/entity/Player.java
++++ b/src/main/java/org/bukkit/entity/Player.java
+@@ -1249,6 +1249,34 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
+      */
+     public void hidePlayer(@NotNull Plugin plugin, @NotNull Player player);
```

Comment:
Default this to `hidePlayer(plugin, player, true)`. Simplify the changes in CraftPlayer accordingly.

Reply:
Not sure I understand the reasoning here? It makes sense to default to true in `CraftPlayer` but not removing this method specifically in the Player interface.

Reply:
```java
default void hidePlayer(Plugin plugin, Player player) {
  hidePlayer(plugin, player, true);
}
```

This removes the implementation of it just defaulting, and lets us see clearly in the API what the default actually is, given the one with the `immediate` param is not actually defaultign anything.

Reply:
TIL default void...

Reply:
default (and static) methods in interfaces are a java 8 feature :P
java 9 added private methods too!